lib/adapters/http/read.js

```diff
@@ -210,6 +224,12 @@ class ReadHTTP extends AdapterRead {
 
                         case 'record':
                             this.pageCount += result.state.total;
+                            break;
+
+                        case 'field':
+                            this.pageCount = (result.state.last_point ? result.state.last_point[this.pageField] : undefined);
```

Comment:
We should trigger a warning if the last point doesn't have the expected field, at least that way the user sees the warning and can figure out why the point is missing the expected field.